### PR TITLE
Test budget squirrel

### PR DESCRIPTION
**Pull request by ianbro**
This is just the set up but I added a couple tests for proof of concept.

### CODE REVIEW

File: server/Tests/BudgetTracker.BudgetSquirrel.Application.Tests/UnitTests/Budgeting/BudgetCreationTests.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace BudgetTracker.BudgetSquirrel.Application.Tests
+{
+    public class BudgetCreationTests
+    {
+        [Fact]
+        public void TestBudgetSavedWhenValidRequestReceived()
+        {
+            BudgetBuilder bb = new BudgetBuilderFactory<Budget>().GetBuilder() as BudgetBuilder;
+            Budget budget = bb.Build();
+            Console.WriteLine(budget.Duration.Id);
```
Comment (BenjaminGroseclose) [inline server/Tests/BudgetTracker.BudgetSquirrel.Application.Tests/UnitTests/Budgeting/BudgetCreationTests.cs]: Doesn't look like you are testing anything here?

File: server/Tests/BudgetTracker.Business.Tests/UnitTests/BudgetCreateTests.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            mockBudgetRepository.Setup(repo => repo.CreateBudget(It.IsAny<Budget>()))
+                                .Returns(Task.FromResult(budget));
+
+            Budget createdBudget = await BudgetCreation.CreateBudgetForUser(budget, owner, mockBudgetRepository.Object);
+
+            mockBudgetRepository.Verify(repo => repo.CreateBudget(budget));
+            Assert.Equal(createdBudget.Name, budget.Name);
+        }
+
+        // (ChildPercentAmount, ChildSetAmount, ParentSetAmount, AmountExpected)
```
Comment (BenjaminGroseclose) [inline server/Tests/BudgetTracker.Business.Tests/UnitTests/BudgetCreateTests.cs]: Haven't seen this used before, love it though. Will this test the test 4 times in a row?

File: server/Tests/BudgetTracker.BudgetSquirrel.Application.Tests/UnitTests/Budgeting/BudgetCreationTests.cs (hunk 1 above)
Comment (ianbro) [reply server/Tests/BudgetTracker.BudgetSquirrel.Application.Tests/UnitTests/Budgeting/BudgetCreationTests.cs]: Oh yeah I gotta delete this. I was debugging the builders

File: server/BudgetTracker.Business/Budgeting/Budget.cs [hunk 3]
```diff
@@ -90,6 +90,11 @@ public bool IsRootBudget
 
         public decimal CalculateBudgetSetAmount()
         {
+            if (!IsRootBudget && ParentBudget == null)
```
[the remark this answers is not in the document]
Comment (ianbro) [reply server/BudgetTracker.Business/Budgeting/Budget.cs]: This is making sure that, if it has a parent budget (parentbudgetid != null), then the actual parent budget object has to be loaded into the ParentBudget property.

File: server/Tests/BudgetTracker.Business.Tests/UnitTests/BudgetCreateTests.cs (hunk 2 above)
Comment (ianbro) [reply server/Tests/BudgetTracker.Business.Tests/UnitTests/BudgetCreateTests.cs]: Yessir!

Comment (BenjaminGroseclose) [reply server/Tests/BudgetTracker.Business.Tests/UnitTests/BudgetCreateTests.cs]: *dabs*

**Pull request merged**